discovery-provider/src/queries/get_challenges.py

```diff
@@ -0,0 +1,127 @@
+from functools import reduce
+from collections import defaultdict
+# pylint: disable=E0611
+from typing import List, DefaultDict, TypedDict, Tuple
+from sqlalchemy import and_
+from src.models import UserChallenge, Challenge, ChallengeType, ChallengeDisbursement
+
+class ChallengeResponse(TypedDict):
+    challenge_id: str
+    user_id: int
+    specifier: str
+    is_complete: bool
+    is_active: bool # need this
+    is_disbursed: bool # need this
+    current_step_count: int
+    max_steps: int
+    challenge_type: str
+
+def rollup_aggregates(user_challenges: List[UserChallenge], challenge: Challenge) -> ChallengeResponse:
```

Comment:
longer term, i think these are the kind of fns we should write unit tests for. it makes it easier from a documentation perspective to see how this reduce works